docs/transaction.md

```diff
@@ -0,0 +1,243 @@
+# Life of a transaction
```

Comment:
It might be useful to start with the definition of Serializability and Linearizability.

https://accelazh.github.io/storage/Linearizability-Vs-Serializability-And-Distributed-Transactions#:~:text=Operations%20in%20Linearizability%20are%20by,there%20are%20no%20concurrenct%20operations.&text=Serializability%20is%20an%20isolation%20level,different%20area%20that%20Linearizability%20originates. 

Serializability

Serializability is an isolation level for database transactions. It comes from database community area, where is a different area that Linearizability originates.

Serializability describes multiple transactions, where a transaction is usually composed of multiple operations on multiple objects.

Database can executed transactions in parallel (and the operations in parallel). Serializability guarantees the result is the same with, as if the transactions were executed one by one. i.e. to behave like executed in a serial order.

Serializability doesn’t guarantee the resulting serial order respects recency. I.e. the serial order can be different from the order in which transactions were actually executed. E.g. Tx1 begins earlier than Tx2, but the result behaves as if Tx2 executed before Tx1. That is also to say, to satisfy the same Serializability, there can be more than one possible execution schedulings.

Reply:
added